File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest = 0
        s = set(nums)
        
        for i in s:
            if i - 1 not in s:
                curr = i 
                curr_streak = 1
                while curr+1 in s:
                    curr+=1
                    curr_streak += 1
                longest = max(longest, curr_streak)
        return longest

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int longest = 0;
        Set<Integer> hashset = new HashSet<Integer>();
        for(int i = 0; i < nums.length; i++){
            hashset.add(nums[i]);
        }
        
        for(Integer num: hashset){
            if (!hashset.contains(num-1)){
                int currLongest = 1;
                int current = num;
                while(hashset.contains(current+1)){
                    current ++;
                    currLongest++;
                }
                longest = Math.max(currLongest, longest);
            }
        }
        return longest;
    }
}